src/test/java/net/logstash/logback/encoder/LogstashEncoderTest.java

```diff
@@ -557,124 +551,123 @@ public void markerIsJSON() throws Exception {
         byte[] encoded = encoder.encode(event);
-        
+
         JsonNode node = MAPPER.readTree(encoded);
-        
+
         assertThat(MAPPER.convertValue(argArray, JsonNode.class).equals(node.get("json_message"))).isEqualTo(true);
     }
-    
+
     @Test
     public void testAppendJsonMessage() throws Exception {
         Object[] argArray = new Object[] { 1, Collections.singletonMap("hello", Collections.singletonMap("hello", "world")) };
         Marker marker = append("json_message", argArray);
         ILoggingEvent event = mockBasicILoggingEvent(Level.INFO);
         when(event.getMarker()).thenReturn(marker);
         when(event.getArgumentArray()).thenReturn(argArray);
-        
+
         encoder.start();
         byte[] encoded = encoder.encode(event);
-        
+
         JsonNode node = MAPPER.readTree(encoded);
-        
+
         assertThat(MAPPER.convertValue(argArray, JsonNode.class).equals(node.get("json_message"))).isEqualTo(true);
     }
-    
+
     @Test
     public void immediateFlushIsSane() {
         encoder.setImmediateFlush(true);
         assertThat(encoder.isImmediateFlush()).isEqualTo(true);
-        
+
         encoder.setImmediateFlush(false);
         assertThat(encoder.isImmediateFlush()).isEqualTo(false);
     }
-    
+
     @Test
     public void includeJsonChunk() throws Exception {
         String customFields = "{\"appname\":\"damnGodWebservice\",\"roles\":[\"customerorder\", \"auth\"], \"buildinfo\": { \"version\" : \"Version 0.1.0-SNAPSHOT\", \"lastcommit\" : \"75473700d5befa953c45f630c6d9105413c16fe1\"} }";
         ILoggingEvent event = mockBasicILoggingEvent(Level.INFO);
-        
+
         encoder.setCustomFields(customFields);
         encoder.start();
         byte[] encoded = encoder.encode(event);
-        
+
         JsonNode node = MAPPER.readTree(encoded);
-        
+
         assertThat(node.get("appname").textValue()).isEqualTo("damnGodWebservice");
         Assert.assertTrue(node.get("roles").equals(parse("[\"customerorder\", \"auth\"]")));
         Assert.assertTrue(node.get("buildinfo").equals(parse("{ \"version\" : \"Version 0.1.0-SNAPSHOT\", \"lastcommit\" : \"75473700d5befa953c45f630c6d9105413c16fe1\"}")));
     }
-    
+
     @Test
     public void customTimeZone() throws Exception {
         final long timestamp = System.currentTimeMillis();
-        
+
         ILoggingEvent event = mockBasicILoggingEvent(Level.ERROR);
         when(event.getTimeStamp()).thenReturn(timestamp);
-        
+
         encoder.setTimeZone("UTC");
         encoder.start();
         byte[] encoded = encoder.encode(event);
-        
+
         JsonNode node = MAPPER.readTree(encoded);
-        
-        assertThat(node.get("@timestamp").textValue()).isEqualTo(FastDateFormat.getInstance("yyyy-MM-dd'T'HH:mm:ss.SSSZZ", TimeZone.getTimeZone("UTC")).format
-                (timestamp));
+
+        assertThat(node.get("@timestamp").textValue()).isEqualTo(String.valueOf(timestamp));
```

Comment:
Same comment as above. Why is the expected value here a long value converted to a string?

Reply:
I missunderstood the StringTimestampWriter, as a result I've done some wrong changes here